insights/parsers/transparent_hugepage.py

```diff
@@ -0,0 +1,85 @@
+"""
+transparent_hugepage sysfs settings
+===================================
+
+Module for parsing the sysfs settings for transparent_hugepage:
+
+ThpUseZeroPage - file ``/sys/kernel/mm/transparent_hugepage/use_zero_page``
+---------------------------------------------------------------------------
+
+Gets the contents of /sys/kernel/mm/transparent_hugepage/use_zero_page, which is either 0 or 1.
+
+Sample input::
+
+    0
+
+Examples:
+
+    >>> shared[ThpUseZeroPage].get()
+    0
+
+
+ThpEnabled - file ``/sys/kernel/mm/transparent_hugepage/enabled``
+-----------------------------------------------------------------
+
+Gets the contents of  /sys/kernel/mm/transparent_hugepage/enabled, which is something like
+`always [madvise] never` where the active value is in brackets.
+
+If no option is active (that should never happen), `get_opt()` will return None.
+
+Sample input::
+
+    always [madvise] never
+
+Examples:
+
+    >>> shared[ThpEnabled].get_line()
+    always [madvise] never
+    >>> shared[ThpEnabled].get_opt()
+    madvise
+
+"""
+
+from .. import Parser, parser
+
+
+@parser("thp_use_zero_page")
+class ThpUseZeroPage(Parser):
+    """
+    Gets the contents of /sys/kernel/mm/transparent_hugepage/use_zero_page, which is either 0 or 1.
+    """
+    def parse_content(self, content):
+        self.data = " ".join(content).strip()
+
+    def get(self):
+        """Returns the string value."""
+        return self.data
```

Comment:
A couple of examples of how you might eliminate the `get` method are:
```
@property
def use_zero_page(self):
    return self.data
```
Alternatively you could set `self.use_zero_page` as an attribute in `parse_content`, and no method would be required.

Reply:
Right. I'll fix it. Thanks.